app/connectors/itsastatus/ITSAStatusUpdateConnector.scala

```diff
@@ -78,11 +78,53 @@ class ITSAStatusUpdateConnector @Inject()(val httpClient: HttpClientV2, val appC
                     .map(failure => s"code: ${failure.code}, reason: ${failure.reason}")
                     .getOrElse("unknown reason")
 
-                log.error(s"response status: ${response.status}, message: $message")
+                logger.error(s"response status: ${response.status}, message: $message")
                 valid
               }
             )
         }
       }
   }
+
+  def makeMultipleItsaStatusUpdateRequests(
+                                            taxYears: List[TaxYear],
+                                            userNino: String,
```

Comment:
Did we ever come up with a solution over what we should do if for say, we update multiple tax years but one happens to fail? It looks like if there's at least one failure itll show an error page but should we be leaving the successful status update as updated?

It may get a bit messy here as if we tried to revert and that failed then it may just end up in a loop.

Reply:
hmm good question yeah having this mechanism for opt out is bit of a headache since some can fail and some could succeed. Guess it would be weird showing the error page when possibly two of the succeeded :l

Reply:
it's not ideal but i feel like we may have to just allow the user to opt out/sign up for the successful year and leave the other one for the user to spot when they get back to the reporting obligations page (feel free to clarify with Chandra however)

In an ideal world the backend would update all the tax years at once and only do the update if all succeed but sadly it doesn't work like that